packages/astro/astro-jsx.d.ts

```diff
@@ -36,164 +36,141 @@ declare namespace astroHTML.JSX {
 	type AstroComponent = import('astro').AstroComponentFactory;
 	type Element = HTMLElement | AstroComponent;
 
-	//
-	// Event Handler Types
-	// ----------------------------------------------------------------------
-	type EventHandler<E extends Event = Event, T extends EventTarget = HTMLElement> = (
-		event: E & { currentTarget: EventTarget & T }
-	) => any;
-
-	type ClipboardEventHandler<T extends EventTarget> = EventHandler<ClipboardEvent, T>;
-	type CompositionEventHandler<T extends EventTarget> = EventHandler<CompositionEvent, T>;
-	type DragEventHandler<T extends EventTarget> = EventHandler<DragEvent, T>;
-	type FocusEventHandler<T extends EventTarget> = EventHandler<FocusEvent, T>;
-	type FormEventHandler<T extends EventTarget> = EventHandler<Event, T>;
-	type ChangeEventHandler<T extends EventTarget> = EventHandler<Event, T>;
-	type KeyboardEventHandler<T extends EventTarget> = EventHandler<KeyboardEvent, T>;
-	type MouseEventHandler<T extends EventTarget> = EventHandler<MouseEvent, T>;
-	type TouchEventHandler<T extends EventTarget> = EventHandler<TouchEvent, T>;
-	type PointerEventHandler<T extends EventTarget> = EventHandler<PointerEvent, T>;
-	type UIEventHandler<T extends EventTarget> = EventHandler<UIEvent, T>;
-	type WheelEventHandler<T extends EventTarget> = EventHandler<WheelEvent, T>;
-	type AnimationEventHandler<T extends EventTarget> = EventHandler<AnimationEvent, T>;
-	type TransitionEventHandler<T extends EventTarget> = EventHandler<TransitionEvent, T>;
-	type MessageEventHandler<T extends EventTarget> = EventHandler<MessageEvent, T>;
-
-	interface DOMAttributes<T extends EventTarget> {
+	interface DOMAttributes {
 		children?: Children;
 
 		// Clipboard Events
-		oncopy?: ClipboardEventHandler<T> | string | undefined | null;
-		oncut?: ClipboardEventHandler<T> | string | undefined | null;
-		onpaste?: ClipboardEventHandler<T> | string | undefined | null;
+		oncopy?: string | undefined | null;
```

Comment:
Is the idea here that an Astro component will always have a string here rather than the actual `EventHandler<T>` object inlined from frontmatter?  Would this break XElement?

Reply:
I doubt it would break XElement, because it doesn't make you write HTML events

However, yes, the following now result in an error in the editor:

```astro
---
function hello() {
	alert("Hey")
}
---

<div onclick={hello}></div>
```

But then again, this code does not work in Astro either - the serialized function you get is wrong, so I'm not sure how to proceed. Either way using TypeScript's DOM types feels weird in Astro, they're much more intended for when you're manipulating events through JavaScript and not when you're writing HTML